src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -903,28 +771,28 @@ func (r *ClusterReconciler) createExternalNodesList(
 	return result, nil
 }
 
-func (r *ClusterReconciler) handleClusterDeletion(
+func (r *ClusterReconciler) removeFinalizers(
 	ctx context.Context, redpandaCluster *vectorizedv1alpha1.Cluster, l logr.Logger,
-) (reconcile.Result, error) {
+) error {
 	log := l.WithName("handleClusterDeletion")
-	log.V(logger.DebugLevel).Info("handling cluster deletion")
+	log.V(logger.DebugLevel).Info("handling finzalizers removal")
 	if controllerutil.ContainsFinalizer(redpandaCluster, FinalizerKey) {
```

Comment:
The logic here could result in a Pod with a left over finalizer because there's no transactions within the k8s API. If we remove the cluster finalizer and `removePodFinalizer` fails on any Pod, Pod finalizers will never be removed.

Instead, we should handle this logic something like:
1. For each Pod and the statefulset
1. If item has a deletion timestamp set and contains any deprecated finalizers
2. Remove the finalizer.

That way we ensure that the finalizers are always removed from any item that could have them. Technically the deletion check is optional but I kinda like to defer changes to the cluster that occur due to operator upgrades when possible. No strong feelings on this particular matter.

Reply:
I agree. 

Slit note that this change does not care about deletion timestamp.